packages/cmf/src/sagaRouter/router.js

```diff
@@ -30,29 +91,39 @@ function shouldRestartSaga(maybeSaga, match) {
 	return false;
 }
 
-function parseSagaState(routeFragments, sagas, currentLocation, index) {
-	const routeFragment = routeFragments[index];
+/**
+ * for a route a list of running saga and current location return a
+ * match object and a saga
+ *
+ * @param {string} routeFragments - the route fragment associated to a saga
+ * @param {RunningTasks} sagas
+ * @param {Location} currentLocation
+ * @param {int} index
+ */
+function parseSagaState(routeFragment, sagas, currentLocation) {
 	return {
-		routeFragment,
 		match: matchPath(currentLocation.pathname, { path: routeFragment }),
 		maybeSaga: sagas[routeFragment],
 	};
 }
 
+/**
+ * responsible to start and cancel saga based on application current url,
+ * restart saga if necessary
+ * @param {object} history - react router history
+ * @param {RoutesConfig} routes
+ */
 export default function* sagaRouter(history, routes) {
 	const sagas = {};
 	const routeFragments = Object.keys(routes);
-	while (true) {  // eslint-disable-line no-constant-condition
+	while (true) {
+		// eslint-disable-line no-constant-condition
```

Comment:
i think this linter will not work very well now :)

Reply:
oups